components/TextControls/examples/supplemental/MDCTextControlTextAreaContentViewController.m

```diff
@@ -0,0 +1,186 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#import "MDCTextControlTextAreaContentViewController.h"
+
+#import "MaterialButtons.h"
+
+#import "MDCBaseTextArea.h"
+#import "MaterialButtons+Theming.h"
+#import "MaterialColorScheme.h"
+
+#import "MDCFilledTextArea+MaterialTheming.h"
+#import "MDCFilledTextArea.h"
+#import "MDCOutlinedTextArea+MaterialTheming.h"
+#import "MDCOutlinedTextArea.h"
+
+@interface MDCTextControlTextAreaContentViewController () <UITextViewDelegate>
+@end
+
+@implementation MDCTextControlTextAreaContentViewController
+
+#pragma mark Setup
+
+- (MDCFilledTextArea *)createMaterialFilledTextArea {
+  MDCFilledTextArea *textArea = [[MDCFilledTextArea alloc] init];
+  textArea.textView.delegate = self;
+  textArea.labelBehavior = MDCTextControlLabelBehaviorFloats;
+  textArea.label.text = @"Phone number";
+  textArea.leadingAssistiveLabel.text = @"This is a string.";
+  [textArea applyThemeWithScheme:self.containerScheme];
+  return textArea;
+}
+
+- (MDCOutlinedTextArea *)createMaterialOutlinedTextArea {
+  MDCOutlinedTextArea *textArea = [[MDCOutlinedTextArea alloc] init];
+  textArea.textView.delegate = self;
+  textArea.label.text = @"Phone number";
+  [textArea applyThemeWithScheme:self.containerScheme];
+  return textArea;
+}
+
+- (MDCBaseTextArea *)createDefaultBaseTextArea {
+  MDCBaseTextArea *textArea = [[MDCBaseTextArea alloc] init];
+  textArea.textView.delegate = self;
+  textArea.label.text = @"This is a floating label";
+  return textArea;
+}
+
+- (void)textViewDidChange:(UITextView *)textView {
+  [self.view setNeedsLayout];
+}
+
+#pragma mark Overrides
+
+- (void)initializeScrollViewSubviewsArray {
+  [super initializeScrollViewSubviewsArray];
+
+  NSArray *textAreaRelatedScrollViewSubviews = @[
+    [self createLabelWithText:@"MDCFilledTextArea:"],
+    [self createMaterialFilledTextArea],
+    [self createLabelWithText:@"MDCOutlinedTextArea:"],
+    [self createMaterialOutlinedTextArea],
+    [self createLabelWithText:@"MDCBaseTextArea:"],
+    [self createDefaultBaseTextArea],
+  ];
+  NSMutableArray *mutableScrollViewSubviews = [self.scrollViewSubviews mutableCopy];
+  self.scrollViewSubviews =
+      [mutableScrollViewSubviews arrayByAddingObjectsFromArray:textAreaRelatedScrollViewSubviews];
+}
+
+- (void)applyThemesToScrollViewSubviews {
+  [super applyThemesToScrollViewSubviews];
+
+  [self applyThemesToTextAreas];
+}
+
+- (void)resizeScrollViewSubviews {
+  [super resizeScrollViewSubviews];
+
+  [self resizeTextAreas];
+}
+
+- (void)enforcePreferredFonts {
```

Comment:
Is this something that we are expecting clients to have to do too? it is not immediately clear that is the case. Let us try and look at this through the lens of a client hunting for code to copy and past into their project.

Concretely we could #pragma mark all the things that clients need to copy?

Reply:
@randallli, The whole appeal of this example is you can tap various buttons to modify the trait collection on the fly to see how the text controls respond to being in dark mode or a larger content size category etc. There is also a toggle error button, which calls either the `-applyThemeWithScheme:` method or the `-applyErrorThemeWithScheme` method. When these methods get called, they always apply the same fonts, regardless of what the content size category is. Unfortunately, there is no way to call theming extension methods that just apply color--they have to apply color _and_ the non-changing typography schemes. So this "enforcePreferredFonts" thing is a workaround. I guess clients would have to do something similar if they wanted to have both dynamic type and the ability to apply error/non-error themes based off user behavior at runtime.

I tried setting `useCurrentContentSizeCategoryWhenApplied` to `YES` on the typography scheme, but that didn't work how I expected it to. Maybe I'm missing a setup step. It's concerning because if I don't know hot to get that property working I'm not sure a client would either. Ideally it'd work out of the box.

Rather than #pragma all the things a client would need to copy I'd rather just have real "typical use" examples. There's one for text fields but not for text areas yet.